src/sidebar/util/fetch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * An error indicating a failed network request.
+ *
+ * Failures that this error can represent include:
+ *
+ *  - Failures to send an HTTP request
+ *  - Requests that returned non-2xx responses
+ *  - Failures to parse the response in the expected format (eg. JSON)
+ */
+export class FetchError extends Error {
+  /**
+   * @param {Response|null} response - The response to the `fetch` request or
+   *   `null` if the fetch failed
+   * @param {string} [reason] - Additional details about the error. This might
+   *   include context of the network request or a server-provided error in
+   *   the response.
+   */
+  constructor(response, reason = '') {
+    let message = 'Network request failed';
+    if (response) {
+      message += ` (${response.status})`;
+    }
+    if (reason) {
+      message += `: ${reason}`;
+    }
+    super(message);
+
+    this.response = response;
+    this.reason = reason;
+  }
+}
+
+/**
+ * Execute a network request and return the parsed JSON response.
+ *
+ * Throws a `FetchError` if making the request fails or the request returns
+ * a non-2xx response.
+ *
+ * Returns `null` if the request returns a 204 (No Content) response.
+ *
+ * @param {string} url
+ * @param {RequestInit} [init] - Parameters for `fetch` request
+ */
+export async function fetchJSON(url, init) {
+  let response;
+  try {
+    response = await fetch(url, init);
+  } catch (err) {
+    // If the request fails for any reason, wrap the result in a `FetchError`.
+    // Different browsers use different error messages for `fetch` failures, so
+    // wrapping the error allows downstream clients to handle this uniformly.
+    throw new FetchError(null, err.message);
+  }
+
+  if (response.status === 204 /* No Content */) {
+    return null;
+  }
+
+  // Attempt to parse a JSON response. This may fail even if the status code
+  // indicates success.
+  let data;
+  try {
+    data = await response.json();
+  } catch (err) {
+    throw new FetchError(response, 'Failed to parse response');
```

Comment:
```suggestion
    throw new FetchError(response, error.message ?? 'Failed to parse response');
```
The above could help to identify what in the response was not parseable, for example:
'Unexpected token < in JSON at position 6'

Reply:
Honestly I don't think the specific error message will be all that useful. From what I've seen in Sentry the most common reason for `response.json()` to fail by far is that the body is missing/empty/unavailable to read. Even with an error like this, you really would need the full response body to make sense of it.